Change to not output not configured warning when renamed and pending cop

ydah opened the pull request:
This PR is change to not output not configured warning when renamed and pending cop.

Resolve: https://github.com/rubocop/rubocop-rspec/issues/1561

## Motivation
This will help eliminate some of the confusion that occurred when we extracted from the Capybara section of RuboCop RSpec to RuboCop Capybara.

RuboCop RSpec 2.18.0 renamed and extracted cop still pending in RuboCop Capybara section.
`RSpec/Capybara/FooCop` -> `Capybara/FooCop`.

If we have the following .rubocop.yml.
```yaml
require:
  - rubocop-capybara
  - rubocop-rspec

Capybara/MatchStyle:
  Enabled: true
Capybara/NegationMatcher:
  Enabled: true
Capybara/SpecificActions:
  Enabled: true
Capybara/SpecificFinders:
  Enabled: true
Capybara/SpecificMatcher:
  Enabled: true
```

If we run RuboCop in this state, we will get the following warning:
```
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file.

Please also note that you can opt-in to new cops by default by adding this to your config:
  AllCops:
    NewCops: enable

Gemspec/DeprecatedAttributeAssignment: # new in 1.30
  Enabled: true
(snip)
RSpec/Capybara/MatchStyle: # new in 2.17
  Enabled: true
RSpec/Capybara/NegationMatcher: # new in 2.14
  Enabled: true
RSpec/Capybara/SpecificActions: # new in 2.14
  Enabled: true
RSpec/Capybara/SpecificFinders: # new in 2.13
  Enabled: true
RSpec/Capybara/SpecificMatcher: # new in 2.12
  Enabled: true
:
For more information: https://docs.rubocop.org/rubocop/versioning.html
```

However, I think it is confusing to see this in the output, since the `RSpec/Capybara` department cop, such as `RSpec/Capybara/MatchStyle`, is actually renamed.

To resolve this, it needs to be added to .rubocop.yml, but it feels a bit strange to explicitly disable the renamed cop.

```yaml
require:
  - rubocop-capybara
  - rubocop-rspec

Capybara/MatchStyle:
  Enabled: true
Capybara/NegationMatcher:
  Enabled: true
Capybara/SpecificActions:
  Enabled: true
Capybara/SpecificFinders:
  Enabled: true
Capybara/SpecificMatcher:
  Enabled: true
RSpec/Capybara:
  Enabled: false
RSpec/Capybara/FeatureMethods: # Moreover, this is not extracted into RuboCop Capybara and must be enabled.
  Enabled: true
```


-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [-] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

pirj reviewed (approved):
Thank you!

Are there any existing specs for those warnings that we can extend to prevent regressions?

bbatsov commented:
I'm fine with the proposed change, but it needs some specs IMO.

ydah commented:
I updated this PR. Thank you for review!

memo) By below file , `RuboCop:: ConfigLoader::warn_on_pending_cops` seems to be suppressed, so I refactored and added specs for easier testing.
https://github.com/rubocop/rubocop/blob/4e7ce1c79809b508cddabbb4b68150150aabdca4/spec/support/suppress_pending_warning.rb#L3-L17
